test/lit_tests/doWhile.mlir

```diff
@@ -0,0 +1,30 @@
+// RUN: enzymexlamlir-opt --canonicalize-scf-for %s | FileCheck %s
+module {
+  func.func @do_while_example() -> index {
+    %c0 = arith.constant 0 : index
+    %c1 = arith.constant 1 : index
+    %c5 = arith.constant 5 : index
+
+    %result = scf.while (%i = %c0) : (index) -> index {
+      // "Before" region (executes first)
+      %updated = arith.addi %i, %c1 : index
```

Comment:
can we do a check here with some generic call(%i, %updated) as well to ensure its set up properly?